### ISSUE TITLE
[Bug]: Template job instance should not be cleaned up to allow KEDA to scale the jobs

### ISSUE THREAD

**Issue by clemlesne** [clemlesne/blue-agent#294]
### Problem

KEDA doesn’t scale jobs after the template job is removed.

Relates to: https://github.com/clemlesne/blue-agent/discussions/290#discussioncomment-10972341

KEDA doc: https://keda.sh/blog/2021-05-27-azure-pipelines-scaler/#placeholder-agent

### Impact

The jobs never run.

### Deployment

Kubernetes with Helm

### Version

9.0.0

**Comment by clemlesne** [comment]
The placeholder agent (I called “template” in the code) should be kept forever.

Template job is already created: https://github.com/clemlesne/blue-agent/blob/80158efe7e6ba6d24b94f4ffa4136d850dcec0e0/src/helm/blue-agent/templates/pod.yaml#L15-L19

But, all agents are made to be cleaned up after run: https://github.com/clemlesne/blue-agent/blob/80158efe7e6ba6d24b94f4ffa4136d850dcec0e0/src/docker/start.sh#L120-L125

What we could implement is to add a flag to ignore the cleanup process for the template instance.

**Comment by Bastichou** [comment]
The following step to make it work manually:
1- Disable the self-agent to prevent any job scheduling on it 
![image](https://github.com/user-attachments/assets/40ce00c0-bde1-489e-86fc-f17742e8f269)
2- I killed the pod from the K8s cluster

**Comment by clemlesne** [comment]
Working on a fix in branch `feat/keda-first-run`.

Changed:

- Added an env var `AZP_TEMPLATE_JOB` to disable the job deletion from the instance pool (thus this first agent will be offline, but still enabled)
- Template jobs are limited to 1 min in execution, enought for run + setup, but no pipeline run, and no more money lost with a compute used for nothing withing the one-hour standard timeout

**Comment by clemlesne** [comment]
@Bastichou, a fix is pushed on `develop` with efa2323243633467f235796dd5a5983e2152c179. Can you try this new buidl and give a feedback?

Integration tests already have been updated but are only testing Bicep and Azure Container Apps at this time. I only built, static tested and manually reviewed the Helm chart.

Thanks!

**Comment by Bastichou** [comment]
Sorry I couldn't had a chance to test it last week! I'll do my best to try it this week :)

**Comment by Bastichou** [comment]
Hi,

I tried to update the Helm chart version to: 9.0.1-27.efa2323 (find it on the build job https://github.com/clemlesne/blue-agent/actions/runs/11467568043/job/31910782351)
But this Helm chart version doesn't exists ?

```
helm search repo clemlesne-blue-agent/blue-agent --versions
NAME                           	CHART VERSION	APP VERSION	DESCRIPTION
clemlesne-blue-agent/blue-agent	9.0.0        	3.244.1    	Deploy Azure Pipelines agent on Kubernetes. Eas...
clemlesne-blue-agent/blue-agent	8.0.0        	3.243.1    	Deploy Azure Pipelines agent on Kubernetes. Eas...
clemlesne-blue-agent/blue-agent	7.1.2        	3.243.1    	Deploy Azure Pipelines agent on Kubernetes. Eas...
clemlesne-blue-agent/blue-agent	7.1.1        	3.243.1    	Deploy Azure Pipelines agent on Kubernetes. Eas...
clemlesne-blue-agent/blue-agent	7.1.0        	3.243.1    	Deploy Azure Pipelines agent on Kubernetes. Eas...
clemlesne-blue-agent/blue-agent	7.0.3        	3.243.1    	Deploy Azure Pipelines agent on Kubernetes. Eas...
clemlesne-blue-agent/blue-agent	7.0.0        	3.234.0    	Deploy Azure Pipelines agent on Kubernetes. Eas...
```

So I tried to run the pipeline with the image version 9.0.1-27.efa2323
Everything worked well, I don't see any regression so far but the agent template is still being used and deleted once the job is done.
I suppose this is because I'm not using the right version of the Helm chart so we are not enjoying the update in the templating.

Cheers,

**Comment by Bastichou** [comment]
I think you only release the helm chart on a push on main branch?

```
  release-helm:
    name: Release Helm chart
    needs:
      - build-release-linux
      - build-release-win
      - build-helm
    # Only deploy on non-scheduled main branch, as there is only one Helm repo and we cannot override an existing version
    if: (github.event_name != 'schedule') && (github.ref == 'refs/heads/main')
```

**Comment by clemlesne** [comment]
Indeed, the Helm chart versions are only deployed from `main` branch.

Can you test installing it from the sources? You can use the commit efa2323243633467f235796dd5a5983e2152c179.

**Comment by bigsem89** [comment]
Is there an update on this?

**Comment by Bastichou** [comment]
I was in holidays the last few weeks.
I'll be able to test it next week :)

**Comment by clemlesne** [comment]
The feature seems to be stabilized in `develop` branch. Will be merged in the next version.

Integration tests now integrate that case.

Feedback appreciated!
